src/lib/object.js

```diff
@@ -0,0 +1,9 @@
+// @ts-check
+
+/**
+ * @param {unknown} value
+ * @returns {value is Record<string, unknown>}
+ */
+export function isRecord(value) {
+  return value !== null && typeof value === 'object' && !Array.isArray(value)
+}
```

Comment:
mixed feelings on introducing this. can `Record<string, unknown>` can be used for class instance types? 

maybe not important given the usage - maybe it's a matter of naming? my understanding is that this is basically narrowing the type to something that can be indexed by a string for the purposes of basic property access

Reply:
Yeah, that's the idea. I named it to match TypeScript's built-in `Record` type (a name I don't personally love).